src/DIRAC/WorkloadManagementSystem/Agent/JobAgent.py

```diff
@@ -324,9 +324,13 @@ def execute(self):
     wholeNode = 'WholeNode' in params
     mpTag = 'MultiProcessor' in params.get('Tags', [])
 
-    if self.extraOptions and '$DIRACROOT' in params.get('Executable', '').strip():
-      params['Arguments'] = (params.get('Arguments', '') + ' ' + self.extraOptions).strip()
-      params['ExtraOptions'] = self.extraOptions
+    if self.extraOptions:
+      if (
+          six.PY2 and '$DIRACROOT' in params.get('Executable', '').strip()
+          or six.PY3 and 'dirac-jobexec' in params.get('Executable', '').strip()
+      ):
```

Comment:
Isn't this checking the Agent's Python version? If so I don't see how that would relate to the params in the job?

Assuming that I'm just missing something it can be written in a slightly cleaner way with:

```suggestion
        if ('$DIRACROOT' if six.PY2 else 'dirac-jobexec') in self.jobArgs.get('Executable', '').strip():
```

Reply:
Ah, that's a good point indeed, the python version of the agent should not make any difference. Correcting it.

Reply:
Does the same apply to `src/DIRAC/WorkloadManagementSystem/JobWrapper/JobWrapper.py`?

Reply:
Indeed! Fixed.